src/mr/worker.go

```diff
@@ -132,11 +150,27 @@ func GetIntermediatePairs(mapf func(string, string) []KeyValue, filename string)
 	}
 
 	kva := mapf(filename, string(content))
-	DPrint("GetIntermediatePairs", len(kva), kva[0])
+	DPrint("GetIntermediatePairs ", len(kva), kva[0])
 	return kva
 }
 
-func InformMapTaskCompletion(intermediateFiles []string) error {
+func InformMapTaskResult(mapTaskId int, intermediateFiles []string, mapf func(string, string) []KeyValue) error {
+	args := InformMapTaskResultArgs{
+		mapTaskId,
+		intermediateFiles,
+	}
+	reply := InformMapTaskResultReply{}
+
+	ok := call("Coordinator.InformMapTaskResult", &args, &reply)
+	if ok {
+		if reply.Action == GetTaskAction {
+			MapWorker(mapf)
```

Comment:
`MapWorker()` is getting called recursively

Possible stack overflow/OOM issue